libcloud/common/worldwidedns.py

```diff
@@ -0,0 +1,200 @@
+        value = "Your limit was exceeded, you need to upgrade your account"
+        super(LimitExceeded, self).__init__(value, http_code, 407,
+                                            driver)
+
+
+class ExistentDomain(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "Domain already exists on our servers"
+        super(ExistentDomain, self).__init__(value, http_code, 408,
+                                             driver)
+
+
+class DomainBanned(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "Domain is listed in DNSBL and is banned from our servers"
+        super(DomainBanned, self).__init__(value, http_code, 409,
+                                           driver)
+
+
+class InvalidDomainName(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "Invalid domain name"
+        super(InvalidDomainName, self).__init__(value, http_code, 410,
+                                                driver)
+
+
+class ErrorOnReloadInNameServer(WorldWideDNSException):
+    def __init__(self, server, http_code, driver=None):
+        if server == 1:
+            value = "Name server #1 kicked an error on reload, contact support"
+            code = 411
+        elif server == 2:
+            value = "Name server #2 kicked an error on reload, contact support"
+            code = 412
+        elif server == 3:
+            value = "Name server #3 kicked an error on reload, contact support"
+            code = 413
+        super(ErrorOnReloadInNameServer, self).__init__(value, http_code, code,
+                                                        driver)
+
+
+class NewUserNotValid(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "New userid is not valid"
+        super(NewUserNotValid, self).__init__(value, http_code, 414,
+                                              driver)
+
+
+class CouldntReachNameServer(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "Couldn't reach the name server, try again later"
+        super(CouldntReachNameServer, self).__init__(value, http_code, 450,
+                                                     driver)
+
+
+class NoZoneFile(WorldWideDNSException):
+    def __init__(self, http_code, driver=None):
+        value = "No zone file in the name server queried"
+        super(NoZoneFile, self).__init__(value, http_code, 451,
+                                         driver)
+
+
+class WorldWideDNSResponse(Response):
+
+    def parse_body(self):
+        """
+        Parse response body.
+
+        :return: Parsed body.
+        :rtype: ``str``
+        """
+        if self._code_response(self.body):
+            codes = self.body.split('\r\n')
+            for code in codes:
+                if code in OK_CODES:
+                    continue
+                elif code in ERROR_CODES:
```

Comment:
An alternative which allows this method to be shorted would be to create a module level dict constant which maps code to the exception class (e.g. `ERROR_CODE_TO_EXCEPTION_CLS`) and using it here.